File: demo/src/main/java/com/ahk/arg/ForSale/models/entities/operations/Alquiler.java
package com.ahk.arg.ForSale.models.entities.operations;

import com.ahk.arg.ForSale.models.entities.persona.Empleado;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter

public class Alquiler extends Operacion{
    private Integer cantMeses;
    @Override
    public Float calcularComisionPara(Empleado empleado) {
        Float comision = (this.cantMeses * super.inmueble.precioFinal()) / 50000;
        empleado.agregarComision(comision);
        return comision;
    }
}

File: entities/alquiler.js
import { Operacion } from "./operacion";

export class Alquiler extends Operacion{
    cantMeses;
    valor_constante;


    constructor(cantMeses, inmueble){
        super("alquiler", inmueble);
        this.cantMeses = cantMeses;
        this.valor_constante = 50000;
        this.comision = calcularComision();
    }

    calcularComision(){ 
        return this.cantMeses * this.inmueble.valor / this.valor_constante;

    }
}